File: engineering/codebase/smartqr.frontend-services/src/domain/codes/content/enums/WifiEncryption.ts
/** Defines the Wi-Fi encryption scheme a network uses — the `T:` field of the WIFI payload. */
export const WifiEncryption = {
  /** WPA / WPA2 / WPA3 personal. */
  Wpa: "wpa",

  /** Legacy WEP. */
  Wep: "wep",

  /** An open network, carrying no password. */
  None: "none",
} as const;

export type WifiEncryption = (typeof WifiEncryption)[keyof typeof WifiEncryption];

File: engineering/codebase/smartqr.backend-services/SmartQr.Common.Domain/Codes/Content/Wifi/Enums/WifiEncryption.cs
namespace SmartQr.Common.Domain.Codes.Content.Wifi.Enums;

/// <summary>Defines the authentication scheme a Wi-Fi network runs.</summary>
public enum WifiEncryption
{
    /// <summary>Represents WPA / WPA2 / WPA3 personal.</summary>
    Wpa,

    /// <summary>Represents legacy WEP.</summary>
    Wep,

    /// <summary>Represents an open network, carrying no password.</summary>
    None,
}
